manifests/config.pp

```diff
@@ -1,7 +1,19 @@
 # == Class chocolatey::config
-#
-# This class is called from chocolatey for service config.
-#
-# NOTE: CURRENTLY NON-FUNCTIONAL
 class chocolatey::config {
+  assert_private()
+
+  #todo: check choco version from custom choco_version fact
+
+  $_choco_exe_path = "${chocolatey::choco_install_location}\\bin\\choco.exe"
+
+  $_enable_autouninstaller = $chocolatey::enable_autouninstaller ? {
+    false => 'disable',
+    default => 'enable'
+  }
+
+  exec { "chocolatey_autouninstaller_${_enable_autouninstaller}":
+    path    => $::path,
+    command => "${_choco_exe_path} feature -r ${_enable_autouninstaller} -n autoUninstaller",
+    unless  => "${_choco_exe_path} feature list -r | findstr /X /I \"autoUninstaller.*\\[${_enable_autouninstaller}d\\]\"",
```

Comment:
@ferventcoder https://tickets.puppetlabs.com/browse/ENTERPRISE-757 is related, though it sounds like that got fixed for Posix.

See http://docs.puppetlabs.com/puppet/latest/reference/resources_exec_windows.html#shell-built-ins. Basically, the Exec provider directly executes commands, rather than passing them to any kind of shell. If you want to use shell built-ins like a pipe, something like cmd has to be invoked directly.